lib/improver/metadata/utilities.py

```diff
@@ -30,10 +30,64 @@
 # POSSIBILITY OF SUCH DAMAGE.
 """General IMPROVER metadata utilities"""
 
+import dask
 import hashlib
+import iris
 # Usage of pickle in this module is only for creation of pickles and hashing
 # the contents. There is no loading pickles which would create security risks.
 import pickle  # nosec
+import numpy as np
+
+
+def create_new_diagnostic_cube(
+        name, units, coordinate_template, attributes=None, data=None):
+    """
+    Creates a template for a new diagnostic cube with suitable metadata.
+
+    Args:
+        name (str):
+            Standard or long name for output cube
+        units (str or cf_units.Unit):
+            Units for output cube
+        coordinate_template (iris.cube.Cube):
+            Cube from which to copy dimensional and auxiliary coordinates
+        attributes (dict or None):
+            Dictionary of attribute names and values
+        data (np.ndarray or None):
+            Data array.  If not set, cube is filled with zeros using a lazy
+            data object, as this will be overwritten later by the caller
+            routine.
+
+    Returns:
+        iris.cube.Cube:
+            Cube with correct metadata to accommodate new diagnostic field
+    """
+    if data is None:
+        data = dask.array.zeros_like(
+            coordinate_template, dtype=np.float32, chunks=-1)
+
+    aux_coords_and_dims, dim_coords_and_dims = [
+        [(coord, coordinate_template.coord_dims(coord))
+         for coord in getattr(coordinate_template, coord_type)]
+        for coord_type in ('aux_coords', 'dim_coords')]
+
+    try:
```

Comment:
I don't think we need this duplication, because `cube.rename` would handle this:
```
cube = iris.cube.Cube(
    data, 
    units=units,
    dim_coords_and_dims=_coords_and_dims(coordinate_template, 'dim_coords'),
    aux_coords_and_dims=_coords_and_dims(coordinate_template, 'aux_coords'),
    attributes=attributes
)
cube.rename(name)
```
...then if we explicitly only want this to be either a standard_name or a long_name (which I don't think we do):
```
if cube.var_name is None:
    raise SomeExceptionHere
```

Reply:
Ooh I like this.  If it works, there's somewhere else in the code I'd like to replace this too.  So may ask you to look at this PR again later and check that.

Reply:
No problem. I did a quick test in the REPL and it seemed to work how I expected.